tests/contrib/kafka/test_kafka.py

```diff
@@ -80,7 +79,10 @@ def empty_kafka_topic(request):
 @pytest.fixture
 def dummy_tracer():
     patch()
-    yield DummyTracer()
+    t = DummyTracer()
+    # disable backoff because it makes these tests less reliable
+    t._writer._send_payload_with_backoff = t._writer._send_payload
```

Comment:
Nice catch! Should we also make this the default behavior for all tests that use the global tracer?

This issue is likely impacting all snapshot tests (or maybe update all/most tests to use the dummy tracer).

Reply:
Trying this out:

https://github.com/DataDog/dd-trace-py/commit/77e924afe0a554c2d652953d5af9245a0988f0a6

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/54570/workflows/57e8fb2b-8f34-4e35-a780-372342834b48

Reply:
It's definitely worth a shot. Based on my testing I'm not highly confident that it will help, but it certainly won't hurt most tests.